examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,573 @@
+]
+
+# %%
+# We can see that the estimator using the `'rbf'` kernel performed best,
+# closely followed by `'linear'`. Both estimators with a `'poly'` kernel
+# performed worse, with the one using a two degree polynomial achieving a much
+# lower perfomance than all other models.
+#
+# Usually, the analysis just ends here, but half the story is missing. **The
+# output of `GridSearchCV` does not provide information on the certainty of
+# the differences between the models. We don't know if these are statistically
+# significant.** To evaluate this, we need to conduct a statistical test.
+# Specifically, to contrast the performance of two models we should
+# statistically compare their AUC scores. There are 100 observations (AUC
+# scores) for each model as we repreated 10 times a 10-fold cross-validation.
+#
+# However, the scores of the models are not independent: we iteratively used
+# the same partitions of the data to evaluate them, increasing the correlation
+# between the performance of the models. This means that some partitions of the
+# data can make the distinction of the classes particularly easy or hard to
+# find for all models, and thus their scores will co-vary.
+#
+# Let's inspect this partition effect by plotting the performance of all models
+# in each fold, and calculating the correlation between models across folds:
+
+# create df of model scores ordered by perfomance
+model_scores = results_df.filter(like='split')
+
+# plot 30 examples of dependency between cv fold and AUC scores
+fig, ax = plt.subplots()
+sns.lineplot(
+    data=model_scores.transpose().iloc[:30],
+    dashes=False, palette='Set1', marker='o', alpha=.5, ax=ax
+)
+ax.set_xlabel("CV test fold", size=12, labelpad=10)
+ax.set_ylabel("Model AUC", size=12)
+ax.tick_params(bottom=True, labelbottom=False)
+plt.show()
+
+# print correlation of AUC scores across folds
+print(f"Correlation of models:\n {model_scores.transpose().corr()}")
+
+# %%
+# We can observe that the performance of the models highly depends on the fold.
+#
+# As a consequence, if we assume independence between observations we will be
+# underestimating the variance computed in our statistical tests, increasing
+# the number of false positive errors (i.e. detecting a significant difference
+# between models when such does not exist) [1]_.
+#
+# Several methods have been developed to correct the variance for the
+# correlation in these cases. In this example we will explore one of these
+# methods implemented using two different statistical approaches: frequentist
+# and Bayesian.
+
+# %%
+# Comparing two models: frequentist approach
+# ------------------------------------------
+#
+# We can start by asking: "Is the first model significantly better than the
+# second model (when ranked by `mean_test_score`)?"
+#
+# To answer this question using a frequentist approach we could
+# run a paired t-test and compute a p-value. Many variants of the latter have
```

Comment:
```suggestion
# run a paired t-test and compute the p-value. This is also known as
# Diebold-Mariano test in the forecast literature. Many variants of this
# t-test have
```

Reply:
Hey, maybe my earlier message was missed, but can you explain here what you mean by this phrase? I couldn't find the corresponding explanation in the papers you mentioned.

Reply:
@martinagvilas Please have a look at [the original paper of Diebold and Mariano](http://www.est.uc3m.es/esp/nueva_docencia/comp_col_get/lade/tecnicas_prediccion/Practicas0708/Comparing%20Predictive%20Accuracy%20(Dielbold).pdf). If you want, we can discuss elsewhere and I can provide way more references if you're interested.

Please just add my suggested sentence.

Reply:
Well, I thought this was exactly the place to have such discussions.

Anyways, I added the phrase but I'm tagging @adrinjalali to make sure this is not out of scope (as mentioned before, the literature defined for this example was something debated for a long time)

Reply:
I'd never heard of Diebold-Mariano test before, but that's fine. I'm happy with having it. I'd also add a reference though. Otherwise LGTM.

Reply:
👍  I added the reference as well then.

Reply:
Thank you. I'll explain later.